Add getting started documents for IDP (#1996)
harness/developer-hub: docs/internal-developer-portal/getting-started/register-a-new-software-component.md

---
title: Add a new Software Component in the catalog

> description: Learn how you can add a new software component in the IDP Software Catalog.

Reviewer: ```suggestion
description: Learn how you can add a new software component in the IDP software catalog.
```

sidebar_position: 30
helpdocs_topic_id:
helpdocs_category_id:
helpdocs_is_private: false
helpdocs_is_published: true
---

Any new Software Component can be registered in the software catalog by creating a `catalog-info.yaml` file in your Git repository and registering its url.

## Create a new `catalog-info.yaml`

If you want to register an existing software component, navigate to its repository. If it is a mono-repo, navigate to its directory and create a `catalog-info.yaml` at the root of the directory. The file can technically live at any place e.g. `.harness/catalog-info.yaml`. Feel free to use this sample YAML -

```yaml
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: my-new-service
  description: Description of my new service
  annotations:
    pagerduty.com/integration-key: <sample-service-integration-key>
  tags:
    - java
  links:
    - url: https://admin.example-org.com
      title: Admin Dashboard
      icon: dashboard
      type: admin-dashboard
spec:
  type: service
  lifecycle: production
  owner: team-a
  system: project-x
```

The key fields for you to update here are

> 

Reviewer: Replace numbering with bullets. This list should be an unordered list.

> 1. `metadata.name` - This should be a unique name of your component, usually the name of the service.

Reviewer: ```suggestion
1. `metadata.name`. This should be a unique name for your component. Usually, it is the name of the service.
```
(Also replace hyphens with periods or colons. I've selected the period here and you can too.)

> 2. `metadata.description` - The description of your new component.

Reviewer: ```suggestion
2. `metadata.description` - A description for your new component.
```

> 3. `spec.type` - The new software component could be a `service`, `library`, `website` or something else.

Reviewer: ```suggestion
3. `spec.type` - The new software component could be a `service`, `library`, `website`, or any other type.
```

4. `spec.owner` - The user group identifier of the team who owns the component.

> Once the file is created on your git, copy the full URL to the file e.g. `https://github.com/harness-community/idp-samples/blob/main/catalog-info.yaml`.

Reviewer: ```suggestion
Once the file is created in your Git repo, copy the full URL to the file. For example, `https://github.com/harness-community/idp-samples/blob/main/catalog-info.yaml`.
```
